app/src/main/java/com/kodeco/android/countryinfo/MainActivity.kt

```diff
@@ -28,7 +31,13 @@ class MainActivity : ComponentActivity() {
 
         setContent {
             MyApplicationTheme {
-                CountryInfoScreen(service)
+                CountryInfoScreen(
+                    viewModel = viewModel(
+                        factory = CountryInfoViewModel.CountryInfoViewModelFactory(
+                            repository = CountryRepositoryImpl(service) ,
+                        ) ,
+                    ) ,
```

Comment:
minor nit: extra spaces before the commas (,)

```suggestion
                            repository = CountryRepositoryImpl(service),
                        ),
                    ),
```